ledger/src/blockstore_db.rs

```diff
@@ -1427,12 +1427,16 @@ pub mod tests {
 
     #[test]
     fn test_excludes_from_compaction() {
-        // currently there are two CFs are excluded from compaction:
-        assert!(excludes_from_compaction(
-            columns::TransactionStatusIndex::NAME
-        ));
-        assert!(excludes_from_compaction(columns::ProgramCosts::NAME));
-        assert!(excludes_from_compaction(columns::TransactionMemos::NAME));
-        assert!(!excludes_from_compaction("something else"));
+        // Column Families are included in compaction by default and must opt out if desired
+        for column_name in Rocks::columns() {
+            match column_name {
+                TransactionStatusIndex::NAME | ProgramCosts::NAME | TransactionMemos::NAME => {
```

Comment:
No compaction means it is equivalent to FIFO compaction without any clean-up logic.

Do you happen to know the reason why we don't do compaction here?

Reply:
The existing comment states this which I'll agree isn't super informative
https://github.com/solana-labs/solana/blob/c7ca2f41f5f763151d9de579b95b8ca589815762/ledger/src/blockstore_db.rs#L359-L360

`ProgramCosts` and `TransactionStatusIndex` are both updated "in-place" (ie the caller that interacts with these deletes or updates values at the same key):
https://github.com/solana-labs/solana/blob/c7ca2f41f5f763151d9de579b95b8ca589815762/core/src/cost_update_service.rs#L202-L222
and
https://github.com/solana-labs/solana/blob/c7ca2f41f5f763151d9de579b95b8ca589815762/ledger/src/blockstore.rs#L2047-L2050

That being said, it isn't clear to me why `TransactionMemos` is on this list ... think I'll need to dig in further